notebook/notebookapp.py

```diff
@@ -1339,11 +1348,14 @@ def init_webapp(self):
     
     @property
     def display_url(self):
-        if self.ip in ('', '0.0.0.0'):
-            ip = socket.gethostname()
+        if self.connect_host:
+            url = self.connect_host + '/'
```

Comment:
Probably don't want to add `+ '/'` here, instead leave `connect_url` untouched.

Reply:
Should `connect_host` be renamed to `custom_connect_host` or `custom_display_host`. I'm a bit confused by the naming.

Reply:
Renamed

@minrk : regarding the trailing '/', i changed things to append it if it is missing. I think it is the "nicest" approach, but tell me if you have reserves on this.